examples/synchronization-mutex-tmpl-level.yaml

```diff
@@ -28,18 +28,18 @@ spec:
 
   - name: acquire-lock
     synchronization:
-      mutex:
-        name: welcome
+      mutexes:
+        - name: welcome
```

Comment:
It might be good to leave the other one in commented out with version specifiers for both. Since people will look at the examples and not know that they have future syntax etc

Reply:
Added as separate examples so that they're:
* Runnable
* Tested

Reply:
I could have been more specific here -- it was a quick initial glance -- but I meant _all_ the examples, including the in-line ones in the docs

> * Tested

This is good to have though.

> Since people will look at the examples and not know that they have future syntax etc

Indeed it looks like this is happening almost as soon as this was merged 😕 https://github.com/argoproj/argo-workflows/issues/13644 plus [two](https://cloud-native.slack.com/archives/C01QW9QSSSK/p1726994955560259) [Slack threads](https://cloud-native.slack.com/archives/C01QW9QSSSK/p1726835793647219?thread_ts=1726835793.647219&cid=C01QW9QSSSK). The issue and former Slack thread are also from a [long](https://github.com/argoproj/argo-workflows/discussions?discussions_q=author%3Ani-todo-spot) [time](https://github.com/argoproj/argo-workflows/issues?q=author%3Ani-todo-spot) user, i.e. not a newbie.

I'm writing up a PR to try to mitigate that

Reply:
Addressed in #13645